core/src/main/java/com/linecorp/armeria/common/ContextAwareEventLoop.java

```diff
@@ -32,18 +36,51 @@
 import io.netty.util.concurrent.ScheduledFuture;
 
 /**
- * A delegating {@link EventLoop} that makes sure all submitted tasks are
- * executed within the {@link RequestContext}.
+ * A delegating {@link EventLoop} that sets the {@link RequestContext} before executing any submitted tasks.
  */
-final class RequestContextAwareEventLoop extends RequestContextAwareExecutorService implements EventLoop {
+public final class ContextAwareEventLoop extends ContextAwareExecutorService implements EventLoop {
```

Comment:
How about extracting the interfaces from the implementations?

Reply:
Fixed. 😉
I didn't extract other package-private classes such as `ContextAwareExecutorService`.
If I need to extract that as well, please let me know.